dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05170DefineFrontEndAndBackEndRoles.java

```diff
@@ -0,0 +1,104 @@
+
+    private static final String ORACLE_RENAME_ROLE_KEY_BACK_END_USER
+            = "UPDATE cms_role SET role_key = 'DOTCMS_BACK_END_USER' WHERE role_key = 'CMS User' ";
+
+
+    private static final String MSSQL_RENAME_ROLE_KEY_FRONT_END_USER
+            = "UPDATE cms_role SET role_key = 'DOTCMS_FRONT_END_USER' , edit_layouts = 0 WHERE role_key = 'LoggedIn Site User' ";
+
+    private static final String MSSQL_RENAME_ROLE_KEY_BACK_END_USER
+            = "UPDATE cms_role SET role_key = 'DOTCMS_BACK_END_USER' WHERE role_key = 'CMS User' ";
+
+
+    private static final String MYSQL_RENAME_ROLE_KEY_FRONT_END_USER
+            = "UPDATE cms_role SET role_key = 'DOTCMS_FRONT_END_USER' , edit_layouts = 0 WHERE role_key = 'LoggedIn Site User' ";
+
+    private static final String MYSQL_RENAME_ROLE_KEY_BACK_END_USER
+            = "UPDATE cms_role SET role_key = 'DOTCMS_BACK_END_USER' WHERE role_key = 'CMS User' ";
+
+    @Override
+    public boolean forceRun() {
+        return true;
+    }
+
+    @Override
+    @CloseDBIfOpened
+    public void executeUpgrade() throws DotDataException, DotRuntimeException {
+        Logger.debug(this, "Upgrading Front-end and Back-End roles definition.");
+        try {
+            DbConnectionFactory.getConnection().setAutoCommit(true);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+
+        final DotConnect dotConnect = new DotConnect();
+
+        dotConnect.setSQL(FRONTEND_ROLE_SANITY_CHECK);
+        final int count = dotConnect.getInt("c");
+        if(count > 0){
+           throw new DotDataException(
+             String.format("The front-end Role defined as `LoggedIn Site User` has %d layouts associated. This must be corrected manually before this upgrade task can run.", count)
+           );
+        }
+
+        if (DbConnectionFactory.isPostgres()) {
+            dotConnect.setSQL(POSTGRES_RENAME_ROLE_KEY_FRONT_END_USER);
+            dotConnect.loadResult();
+
+            dotConnect.setSQL(POSTGRES_RENAME_ROLE_KEY_BACK_END_USER);
+            dotConnect.loadResult();
+        }
+
+        if (DbConnectionFactory.isOracle()) {
+            dotConnect.setSQL(ORACLE_RENAME_ROLE_KEY_FRONT_END_USER);
+            dotConnect.loadResult();
+
+            dotConnect.setSQL(ORACLE_RENAME_ROLE_KEY_BACK_END_USER);
+            dotConnect.loadResult();
+        }
+
+        if (DbConnectionFactory.isMsSql()) {
+            dotConnect.setSQL(MSSQL_RENAME_ROLE_KEY_FRONT_END_USER);
+            dotConnect.loadResult();
+
+            dotConnect.setSQL(MSSQL_RENAME_ROLE_KEY_BACK_END_USER);
+            dotConnect.loadResult();
+        }
+
+        if (DbConnectionFactory.isMySql()) {
+            dotConnect.setSQL(MYSQL_RENAME_ROLE_KEY_FRONT_END_USER);
+            dotConnect.loadResult();
+
+            dotConnect.setSQL(MYSQL_RENAME_ROLE_KEY_BACK_END_USER);
+            dotConnect.loadResult();
+        }
+
```

Comment:
I think we need a startup task that finds all users that have a layout and assign them `DOTCMS_BACK_END_USER` role (unless I have missed it).

Reply:
yeah, I realized I was missing that right after I sent the PR... Already working on it.

Reply:
done.